Automattic/wp-calypso
# Upload a new Gravatar directly from Calypso

This is an in-progress PR for letting users upload a new Gravatar directly from Calypso, solving #7238.

Currently, these are the changes:
1. You can select an image from your computer in `/me`
2. Once an image is selected, it's automatically uploaded to Gravatar.com and set as your new Gravatar
3. The results are logged in the console

![pr-gif](https://cloud.githubusercontent.com/assets/11487924/19088357/72a8b246-8a7e-11e6-9090-a324defba5ba.gif)

Testing is a bit complicated for now :)

**Steps to test on Chrome:**
1. Get a bearer token - you can use [this script](https://gist.github.com/hypest/105aeef7bf863675956c) by @hypest:
   
   ```
   curl -# -X POST --data-urlencode "grant_type=password" --data-urlencode "client_id=appid" --data-urlencode "client_secret=appsecret" --data-urlencode "username=username" --data-urlencode "password=password" https://public-api.wordpress.com/oauth2/token
   ```
   
   You can use Calypso's app ID & secret, found [here](https://github.com/Automattic/wp-calypso/blob/a1c610c300e040a2c101bb5af6a66f7cd72b52d6/config/production.json#L108-L109).
   If you have 2FA enabled, you can add this extra parameter to the call: `--data-urlencode "wpcom_otp=one-time-password"`, or you can create a new application password to use instead.
2. Save the bearer token to localStorage using `localStorage.setItem('bearerToken', 'your_bearer_token')`.
3. Go to `calypso.localhost:3000/me` and click on 'Select Image' to upload a new Gravatar

**Steps to test on Chrome, with oauth enabled:**
1. In `config/development.json`, change `features/oauth` to `true`
2. In `config/_shared.json` set `oauth_client_id` to Calypso's client ID.
3. Go to `calypso.localhost:3000/me` and click on 'Select Image' to upload a new Gravatar

**Steps to test on the Desktop app:**

The desktop app does use a bearer token for making wpcom requests, so it's a bit easier to test.
1. Install the desktop app [as explained here](https://github.com/Automattic/wp-desktop)
2. Inside the `wp-desktop/calypso` directory, make sure to checkout the `add/gravatar-uploading` branch
3. In the file `desktop-config/config-base.json`, add `"webSecurity": false` under `webPreferences`
4. Run the desktop app
5. One the app is running, go to `/me`and and click on 'Select Image' to upload a new Gravatar
